test/trivial_general.js

```diff
@@ -1,17 +1,43 @@
-var TrivialToken = artifacts.require("TrivialToken.sol");
+var DevelopmentToken = artifacts.require("DevelopmentTrivialToken.sol");
 
 
 contract('TrivialToken - General tests', (accounts) => {
 
     var token;
 
     beforeEach(async () => {
-        token = await TrivialToken.deployed();
+        token = await DevelopmentToken.new(
+            'TrivialTest',
+            'TRVLTEST',
+            Math.floor(Date.now() / 1000 + 600),
+            600,
+            accounts[8],
+            accounts[9],
+            200000,
+            100000,
+            700000
+        );
     })
 
+    async function throws(fn, ...args) {
+        thrown = false;
+        try { await fn(...args); }
+        catch (err) { thrown = true; }
+        return thrown;
+    }
+
     it('should create TrivialToken', async () => {
-        assert.equal(await token.name.call(), 'TrivialTest', 'Token name is not Trivial')
-        assert.equal(await token.currentState.call(), 0, 'Current state is different')
+        assert.equal(await token.name.call(), 'TrivialTest', 'Token name is not Trivial');
+        assert.equal(await token.currentState.call(), 0, 'Current state is different');
+    })
+
+    it('should test selfdestruct', async () => {
+        assert.equal(await token.name.call(), 'TrivialTest', 'Token name is not Trivial');
+        await token.becomeTrivial();
+        await token.setStateIcoCancelled();
+        assert.equal(await token.currentState.call(), 5, 'Current state is different');
```

Comment:
1. The problem here is that I have no idea what this 5 means here. It may mean AuctionFinished state and nobody would notice that tests are wrong. I would create State enum object where all states are described and would reference it here like this: `State.IcoCancelled`.
2. Another question is if that assert is really necessary. You called `setStateIcoCancelled` so it must be in this state now. If you want to test if `setStateIcoCancelled` is working correctly then you should create separate test for that, but I don't think this is necessary.